syntax = "proto2";
import "goby/acomms/protobuf/driver_base.proto";
import "dccl/option_extensions.proto";

package goby.moos.bluefin.protobuf;

message HardwareRatePair
{
    required string hardware_name = 1;
    required int32 rate = 2;
    required int32 packet_bytes = 3;
}

message BluefinModemIdLookUp
{
    required int32 goby_id = 1;
    required int32 bluefin_id = 2;
}

message Config
{
    optional string moos_server = 1 [default = "localhost"];
    optional int32 moos_port = 2 [default = 9000];

    optional string nmea_in_moos_var = 3 [default = "IFS_RAW_IN"];
    optional string nmea_out_moos_var = 4 [default = "IFS_RAW_OUT"];

    repeated HardwareRatePair hardware_to_rate = 5;

    repeated BluefinModemIdLookUp modem_lookup = 6;
}

extend goby.acomms.protobuf.DriverConfig
{
    optional Config config = 1361;
}
